src/Platform/Microsoft.Testing.Platform/Configurations/AggregatedConfiguration.cs

```diff
@@ -64,7 +70,11 @@ public void SetTestHostWorkingDirectory(string workingDirectory)
         _testHostWorkingDirectory = workingDirectory;
     }
 
-    public async Task CheckTestResultsDirectoryOverrideAndCreateItAsync(ICommandLineOptions commandLineOptions, IRuntime runtime, ServiceProvider serviceProvider)
+    public async Task CheckTestResultsDirectoryOverrideAndCreateItAsync(
+        ICommandLineOptions commandLineOptions,
+        IRuntime runtime,
+        IFileSystem fileSystem,
+        ServiceProvider serviceProvider)
```

Comment:
@cvpoienaru @MarcoRossignoli Can't we have these parameters as ctor instead? It feels weird to have technical object on a public API.

Reply:
api is not public this object is internal or? it's needed because if I recall well we have these object late and not when we create the type, if are available before fine to me.

Reply:
So what should we do about this method ? Since the type is not public, should we address it later ?

Reply:
We discussed off-line with @Evangelink and for now it can stay as-is, we will refactor if we find a good way to not confuse the "state" and the references mutation/aliasing.

Reply:
Great, thanks ! Will resolve the conversation then.